### PR TITLE
A Dockerfile for Strelka 1.0.15

### PR DESCRIPTION
**Pull request by tmooney**
This PR is against the "1.0.15" branch.  Based on [these docs](https://docs.docker.com/docker-hub/builds/) it seems like we should have the "master" branch point to whatever the latest version is, so I'd suggest we merge this to be the current master if it's accepted to the 1.0.15 branch.

### CODE REVIEW

File: docker_helper.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+TUMOR_BAM=$1
+NORMAL_BAM=$2
+REFERENCE=$3
+OUTPUT_DIR=$4
+
+$STRELKA_INSTALL_DIR/bin/configureStrelkaWorkflow.pl --tumor=$TUMOR_BAM --normal=$NORMAL_BAM --ref=$REFERENCE --output-dir=$OUTPUT_DIR --config=$STRELKA_INSTALL_DIR/etc/strelka_config_bwa_default.ini
+
+cd $OUTPUT_DIR
+make -j 4
```
Comment (jasonwalker80) [inline docker_helper.sh]: I had been using 8 jobs when running this manually.  I believe that was a number recommended by the strelka author as well.

File: docker_helper.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    echo "Usage: $0 [TUMOR_BAM] [NORMAL_BAM] [REFERENCE] [OUTPUT_DIR]"
+    exit 255
+fi
+
+TUMOR_BAM=$1
+NORMAL_BAM=$2
+REFERENCE=$3
+OUTPUT_DIR=$4
+
+$STRELKA_INSTALL_DIR/bin/configureStrelkaWorkflow.pl --tumor=$TUMOR_BAM --normal=$NORMAL_BAM --ref=$REFERENCE --output-dir=$OUTPUT_DIR --config=$STRELKA_INSTALL_DIR/etc/strelka_config_bwa_default.ini
```
Comment (jasonwalker80) [inline docker_helper.sh]: The config will only work for BWA aligned WGS.  For Exome the config needs to be updated like so `s/isSkipDepthFilters = 0/isSkipDepthFilters = 1/`.  Should the config file be another input to allow for configuration, ex aligner and data type?

Comment (tmooney) [inline docker_helper.sh]: I'd prefer to expose a config option and not embed business logic into this docker image.  Maybe we leave the `strelka_config_bwa_default.ini` as a default, or maybe we require one to always be specified?

Comment (jasonwalker80) [inline docker_helper.sh]: Yeah, the sed-like substitution above is an example of the parameter and how the value is updated.  I think it's fine to require a config as input.  The config files are shipped with the strelka code as a convenience.  It doesn't seem unreasonable to ask a user of this image to store that config and provide it as an input.  In the same way, should we add an optional input for [extraStrelkaArguments](https://sites.google.com/site/strelkasomaticvariantcaller/home/configuration-and-analysis/configuration-extrastrelkaoptions)

Comment (jasonwalker80) [inline docker_helper.sh]: Scratch the extraStrelkaArguments question as it seems those can be defined in the config.  Adding the config input would be all that is required.

**Comment by jasonwalker80** [comment]
+2

**Pull request merged**